src/hotspot/share/prims/resolvedMethodTable.cpp

```diff
@@ -56,7 +56,7 @@ unsigned int method_hash(const Method* method) {
   unsigned int hash = method->klass_name()->identity_hash();
   hash = (hash * 31) ^ method->name()->identity_hash();
   hash = (hash * 31) ^ method->signature()->identity_hash();
-  return hash;
+  return hash ^ (unsigned)((uintptr_t)method >> (LogMinObjAlignmentInBytes + 3));
```

Comment:
I agree with Vladimir that it makes sense to improve the hashing function of the ResolvedMethodTable for the special case of names matching.  You have to use the ClassLoaderData though, not the Method* because the reason for the table is to replace Method* in the case of redefinition.

Reply:
fixed, thanks.